packages/dx-react-grid-export/src/plugins/grid-exporter-core.tsx

```diff
@@ -0,0 +1,191 @@
+      filters, sorting, selection, totalSummaryItems, groupSummaryItems, customizeSummaryCell,
+    } = this.props;
+
+    const workbook = createWorkbook();
+    const worksheet = createWorksheet(workbook);
+
+    const summaryExists = totalSummaryItems || groupSummaryItems;
+    const useGrouping = !!grouping?.length;
+    const useSelection = !!selection;
+    const useFilters = !!filters;
+    const useSorting = !!sorting;
+    const useColumnOrder = !!columnOrder;
+    const useColumnVisibility = !!hiddenColumnNames;
+
+    return (
+      <Plugin>
+        <GridCoreGetters
+          rows={propRows}
+          columns={propColumns}
+          getRowId={getRowId}
+          getCellValue={getCellValue}
+        />
+        <TableColumnsWithDataRowsGetter
+          columnExtensions={columnExtensions}
+        />
+        {useColumnVisibility && (
+          <VisibleTableColumns hiddenColumnNames={hiddenColumnNames} />
+        )}
+        {useColumnOrder && (
+          <OrderedTableColumns order={columnOrder} />
+        )}
+        <Getter name="isExporting" value />
+
+        {/* State */}
+        {useFilters && (
+          <FilteringState filters={filters} />
+        )}
+        {useSorting && (
+          <SortingState sorting={sorting} />
+        )}
+        {useGrouping && (
+          <GroupingState grouping={grouping} />
+        )}
+        {summaryExists && (
+          <SummaryState totalItems={totalSummaryItems} groupItems={groupSummaryItems} />
+        )}
+        {useSelection && (
+          <SelectionState selection={selection} />
+        )}
+
+        {/* Integrated */}
+        {useGrouping && (
+          <TableColumnsWithGrouping
+            columnExtensions={groupColumnExtensions}
+            showColumnsWhenGrouped={showColumnsWhenGrouped}
+          />
+        )}
+        {useFilters && (
+          <IntegratedFiltering />
+        )}
+        {useSorting && (
+          <IntegratedSorting />
+        )}
+        {useGrouping && (
+          <IntegratedGrouping />
+        )}
+        {summaryExists && (
+          <IntegratedSummary />
+        )}
+
+        <Getter name="customizeSummaryCell" value={customizeSummaryCell} />
+        <Getter name="workbook" value={workbook} />
+        <Getter name="worksheet" value={worksheet} />
+        <Getter name="maxGroupLevel" computed={maxGroupLevelComputed} />
+        <Getter name="outlineLevels" computed={outlineLevelsComputed} />
+        <Getter name="rows" computed={rowsToExportComputed} />
+        <Getter name="groupTree" computed={groupTreeComputed} />
+        <Getter name="exportSummary" computed={exportSummaryComputed} />
+        <Getter name="getCloseGroup" computed={getCloseGroupComputed} />
+
+        <Action name="performExport" action={this.performExport} />
+        <Template name="root">
+          {() => (
+            <TemplateConnector>
+              {(_, { performExport }) => {
+                performExport();
```

Comment:
Why do we call something in the render method? I think `componentDidUpdate` is the better place for it

Reply:
We should have an access to getters and actions in `perfromExport` method. I didn't find any other solution